File: doc-examples/example-kotlin/src/main/kotlin/io/micronaut/gcp/pubsub/client/SimpleClient.kt
package io.micronaut.gcp.pubsub.client

//tag imports[]
import com.google.pubsub.v1.PubsubMessage
import io.micronaut.gcp.pubsub.annotation.PubSubClient
import io.micronaut.gcp.pubsub.annotation.Topic
import io.micronaut.gcp.pubsub.support.Animal
// end::imports[]

// tag::clazz[]
@PubSubClient // <1>
interface SimpleClient {

	@Topic("animals")
	fun send(message: PubsubMessage) // <2>

	@Topic("animals")
	fun send(data: ByteArray) // <3>

	@Topic("animals")
	fun send(animal: Animal) // <4>
}
// end::clazz[]

File: test-suite-python/src/main/python/micronaut/gcp/pubsub/client/SimpleClient.py
from abc import ABC, abstractmethod

# tag::imports[]
from com.google.pubsub.v1 import PubsubMessage
from micronaut.gcp.pubsub.annotation import PubSubClient, Topic
from micronaut.gcp.pubsub.support.Animal import Animal
# end::imports[]


# tag::clazz[]
@PubSubClient  # <1>
class SimpleClient(ABC):

    @Topic("animals")
    @abstractmethod
    def send_message(self, message: PubsubMessage) -> None:  # <2>
        ...

    @Topic("animals")
    @abstractmethod
    def send_bytes(self, data: bytes) -> None:  # <3>
        ...

    @Topic("animals")
    @abstractmethod
    def send_animal(self, animal: Animal) -> None:  # <4>
        ...
# end::clazz[]
